mantidproject/mantid
# Phase Table Validation In Muon Analysis

**Description of work.**
Added some validation to first good data and last good data with some unit tests. There was an error message that occurred when trying to create a table whilst outputting fit information. The message occurred as workspace names were being overwritten when grouped together. To avoid this for now I have stopped the grouping for fit information workspace. The fix for calculating with an empty table shouldn't be a problem given the tab is disabled now when no data is loaded which was a fix from #30181 

**To test:**
1. Open Muon Analysis
2. Load a run (e.g. MUSR62260)
3. Go to phase table tab
4. Check calculate table still works as before
5. Now try change first good and last good data
    1. You should not be able to make first > last
    2. first < min or > max
    3. last < min or > max
6. Check output fit information
7. then calculate table. this should not produce an error message as it did before, and the workspace should be in the ADS on it's own
8. Check loading other runs and outputting fit information also works  

Fixes #30168 



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
